File: 1164.cpp
#include <iostream>

using namespace std;

int main(void) {
    int q, i, n, s, j;
    cin >> q;
    for (i = 0; i < q; ++i) {
        cin >> n;
        s = 0;
        for (j = 1; j < n / 2 + 1; ++j)
            if (n % j == 0) s += j;
        if (n == s) cout << n << " eh perfeito" << endl;
        else cout << n << " nao eh perfeito" << endl;
    }
    return 0;
}

File: 1-iniciante/1164.py
for g in range(int(input())):
    n = int(input())
    lim = int(n / 2) + 1
    s = 0
    for i in range(1, lim):
        if n % i == 0: s += i
    if s == n: print(n, 'eh perfeito')
    else: print(n, 'nao eh perfeito')
